python/cudf/cudf/core/column/categorical.py

```diff
@@ -26,6 +27,11 @@
     min_unsigned_type,
 )
 
+
+def custom_weakref_callback(ref):
+    pass
```

Comment:
Here is why we need it:

```python
    # A callback for ``weakref.ref`` API to generate unique
    # weakref instances that can be counted correctly.

    # Example below shows why this is necessary:

    In [1]: import cudf
    In [2]: import weakref

    # Let's create an object ``x`` that we are going to weakref:

    In [3]: x = cudf.core.buffer.BufferWeakref(1, 2)

    # Now generate three weak-references of it:

    In [4]: a = weakref.ref(x)
    In [5]: b = weakref.ref(x)
    In [6]: c = weakref.ref(x)

    # ``weakref.ref`` actually returns the same singleton object:

    In [7]: a
    Out[7]: <weakref at 0x7f5bea052400; to 'BufferWeakref' at 0x7f5c99ecd850>
    In [8]: b
    Out[8]: <weakref at 0x7f5bea052400; to 'BufferWeakref' at 0x7f5c99ecd850>
    In [9]: c
    Out[9]: <weakref at 0x7f5bea052400; to 'BufferWeakref' at 0x7f5c99ecd850>

    In [10]: a is b
    Out[10]: True
    In [11]: b is c
    Out[11]: True

    # This will be problematic as we cannot determine what is the count
    # of weak-references:

    In [12]: weakref.getweakrefcount(x)
    Out[12]: 1

    # Notice, though we want ``weakref.getweakrefcount`` to return ``3``, it
    # returns ``1``. So we need to work-around this by using an empty/no-op
    # callback:

    In [13]: def custom_weakref_callback(ref):
        ...:     pass
        ...:


    In [14]: d = weakref.ref(x, custom_weakref_callback)
    In [15]: e = weakref.ref(x, custom_weakref_callback)
    In [16]: f = weakref.ref(x, custom_weakref_callback)

    # Now there is a unique weak-reference created:

    In [17]: d
    Out[17]: <weakref at 0x7f5beb03e360; to 'BufferWeakref' at 0x7f5c99ecd850>
    In [18]: e
    Out[18]: <weakref at 0x7f5bd15e3810; to 'BufferWeakref' at 0x7f5c99ecd850>
    In [19]: f
    Out[19]: <weakref at 0x7f5bd15f1f40; to 'BufferWeakref' at 0x7f5c99ecd850>

    # Now calling ``weakref.getweakrefcount`` will result in ``4``, which is correct:

    In [20]: weakref.getweakrefcount(x)
    Out[20]: 4

    In [21]: d is not e
    Out[21]: True

    In [22]: d is not f
    Out[22]: True

    In [23]: e is not f
    Out[23]: True
```